gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/CheckPegasusSnapshotTask.java

```diff
@@ -1,32 +1,176 @@
+        getProject().fileTree(_currentSnapshotDirectory), getProject().fileTree(_previousSnapshotDirectory), false));
+    if (argFiles.isEmpty())
+    {
+      return;
+    }
+
+    FileCollection pathedCodegenClasspath;
+    try
+    {
+      pathedCodegenClasspath = PathingJarUtil.generatePathingJar(getProject(), getName(),
+          _codegenClasspath, false);
+    }
+    catch (IOException e)
+    {
+      throw new GradleException("Error occurred generating pathing JAR.", e);
+    }
+
+    CompatibilityLogChecker logChecker = new CompatibilityLogChecker();
+    getProject().javaexec(javaExecSpec ->
+    {
+      javaExecSpec.setMain("com.linkedin.restli.tools.snapshot.check.PegasusSchemaSnapshotCompatibilityChecker");
+      javaExecSpec.setClasspath(pathedCodegenClasspath);
+      javaExecSpec.args("--compatLevel", _compatibilityLevel);
+      javaExecSpec.args("--compatMode", _compatibilityMode);
+      javaExecSpec.args("--report");
+      javaExecSpec.args(argFiles);
+      javaExecSpec.setStandardOutput(logChecker);
+    });
+
+    IOUtil.writeText(_summaryTarget, logChecker.getWholeText());
+
+    if (!logChecker.isModelCompatible())
+    {
+      throw new GradleException("See output for " + getPath() + ". Summary written to "
+          + _summaryTarget.getAbsolutePath());
+    }
+  }
+
+  @InputDirectory
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  public File getCurrentSnapshotDirectory()
+  {
+    return _currentSnapshotDirectory;
   }
 
+  public void setCurrentSnapshotDirectory(File currentSnapshotDirectory)
+  {
+    _currentSnapshotDirectory = currentSnapshotDirectory;
+  }
+
+  @InputDirectory
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  public File getPreviousSnapshotDirectory()
+  {
+    return _previousSnapshotDirectory;
+  }
+
+  public void setPreviousSnapshotDirectory(File previousSnapshotDirectory)
+  {
+    _previousSnapshotDirectory = previousSnapshotDirectory;
+  }
+
+  @Input
+  public String getCompatibilityLevel()
+  {
+    return _compatibilityLevel;
+  }
+
+  public void setCompatibilityLevel(String compatibilityLevel)
+  {
+    _compatibilityLevel = compatibilityLevel;
+  }
+
+  @Input
+  public String getCompatibilityMode()
+  {
+    return _compatibilityMode;
+  }
+
+  public void setCompatibilityMode(String compatibilityMode)
+  {
+    _compatibilityMode = compatibilityMode;
+  }
+
+  @Classpath
+  public FileCollection getCodegenClasspath()
+  {
+    return _codegenClasspath;
+  }
+
+  public void setCodegenClasspath(FileCollection codegenClasspath)
+  {
+    _codegenClasspath = codegenClasspath;
+  }
+
+  @Input
+  public boolean isExtensionSchema()
+  {
+    return _isExtensionSchema;
+  }
+
+  public void setExtensionSchema(boolean isExtensionSchema)
+  {
+    _isExtensionSchema = isExtensionSchema;
+  }
+
+  @Classpath
+  public FileCollection getHandlerJarPath()
+  {
+    return _handlerJarPath;
```

Comment:
unused?

Reply:
It will be used to in the following PR. I created it as for a place holder.